plugins/processors/awsapplicationsignals/internal/normalizer/attributesnormalizer.go

```diff
@@ -9,47 +9,63 @@ import (
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	conventions "go.opentelemetry.io/collector/semconv/v1.18.0"
+	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
 	"go.uber.org/zap"
 
-	attr "github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsappsignals/internal/attributes"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/common"
+	attr "github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/internal/attributes"
+)
+
+const (
+	MaxEnvironmentLength         = 259
+	MaxServiceNameLength         = 255
+	DefaultMetricAttributeLength = 1024
 )
 
 type attributesNormalizer struct {
 	logger *zap.Logger
 }
 
-var renameMapForMetric = map[string]string{
-	attr.AWSLocalService:    "Service",
-	attr.AWSLocalOperation:  "Operation",
-	attr.AWSRemoteService:   "RemoteService",
-	attr.AWSRemoteOperation: "RemoteOperation",
-	attr.AWSRemoteTarget:    "RemoteTarget",
+var attributesRenamingForMetric = map[string]string{
+	attr.AWSLocalService:             common.MetricAttributeLocalService,
+	attr.AWSLocalOperation:           common.MetricAttributeLocalOperation,
+	attr.AWSLocalEnvironment:         common.MetricAttributeEnvironment,
+	attr.AWSRemoteService:            common.MetricAttributeRemoteService,
+	attr.AWSRemoteOperation:          common.MetricAttributeRemoteOperation,
+	attr.AWSRemoteEnvironment:        common.MetricAttributeRemoteEnvironment,
+	attr.AWSRemoteTarget:             common.MetricAttributeRemoteResourceIdentifier,
+	attr.AWSRemoteResourceIdentifier: common.MetricAttributeRemoteResourceIdentifier,
+	attr.AWSRemoteResourceType:       common.MetricAttributeRemoteResourceType,
 }
 
-var renameMapForTrace = map[string]string{
+var resourceAttributesRenamingForTrace = map[string]string{
 	// these kubernetes resource attributes are set by the openTelemetry operator
 	// see the code references from upstream:
 	// * https://github.com/open-telemetry/opentelemetry-operator/blob/0e39ee77693146e0924da3ca474a0fe14dc30b3a/pkg/instrumentation/sdk.go#L245
 	// * https://github.com/open-telemetry/opentelemetry-operator/blob/0e39ee77693146e0924da3ca474a0fe14dc30b3a/pkg/instrumentation/sdk.go#L305C43-L305C43
-	attr.K8SDeploymentName:  "K8s.Workload",
-	attr.K8SStatefulSetName: "K8s.Workload",
-	attr.K8SDaemonSetName:   "K8s.Workload",
-	attr.K8SJobName:         "K8s.Workload",
-	attr.K8SCronJobName:     "K8s.Workload",
-	attr.K8SPodName:         "K8s.Pod",
+	semconv.AttributeK8SDeploymentName:  "K8s.Workload",
+	semconv.AttributeK8SStatefulSetName: "K8s.Workload",
+	semconv.AttributeK8SDaemonSetName:   "K8s.Workload",
+	semconv.AttributeK8SJobName:         "K8s.Workload",
+	semconv.AttributeK8SCronJobName:     "K8s.Workload",
+	semconv.AttributeK8SPodName:         "K8s.Pod",
+}
+
+var attributesRenamingForTrace = map[string]string{
+	attr.AWSRemoteTarget: attr.AWSRemoteResourceIdentifier,
```

Comment:
Q: I think we will also change `RemoteTarget` to `RemoteResourceIdentifier` in SDK directly, right?

Reply:
Yes, Check this. https://github.com/aws-observability/aws-otel-java-instrumentation/pull/786